waflib/Task.py

```diff
@@ -1080,7 +1080,7 @@ def replc(m):
 			if meth:
 				app('tsk.inputs%s' % meth)
 			else:
-				app('" ".join([a.path_from(cwdx) for a in tsk.inputs])')
+				app('" ".join([a.abspath() for a in tsk.inputs])')
```

Comment:
there are some problems with absolute paths:
 - it mucks up ccache when you go between two different directories on your machine
 - long filenames don't work in cygwin (there is quite a short absolute path limit) and absolute paths are much more likely to be over the limit
 - if you rename a git checkout dir then abs paths break

Reply:
The cygwin is a valid case but the others are easily solved with a ./waf clean or distclean.